pkg/piecestore/psserver/agreementsender/agreementsender_test.go

```diff
@@ -0,0 +1,118 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package agreementsender_test
+
+import (
+	"crypto/rand"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap/zaptest"
+
+	"storj.io/storj/internal/memory"
+	"storj.io/storj/internal/testcontext"
+	"storj.io/storj/internal/testplanet"
+	"storj.io/storj/pkg/storj"
+)
+
+func TestSendAgreementsToSatellite(t *testing.T) {
+	ctx := testcontext.New(t)
+	defer ctx.Cleanup()
+
+	planet, err := testplanet.NewCustom(zaptest.NewLogger(t), testplanet.Config{
+		SatelliteCount: 1, StorageNodeCount: 10, UplinkCount: 1,
+	})
+	require.NoError(t, err)
+	defer ctx.Check(planet.Shutdown)
+
+	planet.Start(ctx)
+
+	before := time.Now()
+	for _, node := range planet.StorageNodes {
+		node.Agreements.Sender.Loop.Pause()
+	}
+
+	data := make([]byte, 500*memory.KiB)
+	_, err = rand.Read(data)
+	assert.NoError(t, err)
+
+	err = planet.Uplinks[0].Upload(ctx, planet.Satellites[0], "test/bucket", "test/path/first", data)
+	assert.NoError(t, err)
+
+	putAllocation := make(map[storj.NodeID]int64)
+	for _, node := range planet.StorageNodes {
+		allocations, err := node.DB.PSDB().GetBandwidthAllocations()
+		assert.NoError(t, err)
+
+		allocPerSat := allocations[planet.Satellites[0].ID()]
+		if len(allocPerSat) > 0 {
+			assert.Equal(t, 1, len(allocPerSat))
+			putAllocation[node.ID()] = allocPerSat[0].Agreement.Total
+		}
+	}
+
+	for _, node := range planet.StorageNodes {
+		node.Agreements.Sender.Loop.Trigger()
+	}
+
+	time.Sleep(1 * time.Second)
```

Comment:
Is this sleep necessary? There are quite a few methods on cycle that allows for it to be completed.

Reply:
it fails locally without Sleep so most probably it will fail on travis

Reply:
Use `TriggerWait`.

Reply:
ah I misunderstand earlier whats `TriggerWait` is doing. Now its working without sleep. Thanks